Python/Product/TestAdapter.Executor/PythonFiles/testing_tools/adapter/report.py

```diff
@@ -72,3 +72,44 @@ def report_discovered(tests, parents, pretty=False, simple=False,
     serialized = json.dumps(data, **kwargs)
 
     _send(serialized)
+
+
+
+
+
+
+def report_unittest_discovered(suites, parents, pretty=False, simple=False,
+                      _send=print, **_ignored):
+    """Serialize the discovered tests and write to stdout."""
+    data = []
+        
+    for suite in suites._tests:
+        root = {
+            'tests': []
+            }
+        for cls in suite._tests:
+            try:
+                for test in cls._tests:
+                    testdata = {
+                        'id': test.id(),
+                        'name': test._testMethodName,
+                        'lineno' : test.lineno,
+                        'source' : test.source
+                        }
```

Comment:
currently these are the attributes i'm pulling from unittest discovery.

test.id() and _testMethodName came from unittests's testloader.. i had to add lineno and source in unittest/_discover.py below